### DISCUSSION TITLE
suggestion: automating a link to openstreetmap

### DISCUSSION THREAD

**Discussion by backache** [openbenches/openbenches.org#322] [Ideas]
Given that what we find are benches and we have geographic co-ordinates for them, can this data be matched with the record of the bench in openstreetmap and the openbenches ID gets pushed into  Key:openbenches:id

**Comment by edent** [comment]
I'm not familiar with the OSM API. I'm happy to take a Pull Request if you can work out how to match our IDs with theirs.

**Comment by backache** [comment]
I'll do some digging and if I can get a sensible answer I'll come back. As you may have guessed I am really into the idea that "no data should be an island" and think interconnectivity increases impact.

@jrbray1 you mentioned OSM before, do you have any insights into how this would work?

I have asked OSM on their community site

https://help.openstreetmap.org/questions/84640/how-to-do-automated-addition-of-data-to-osm-from-a-crowdsourcing-site

**Comment by jrbray1** [comment]
To be honest I don't think linking OB and OSM would work, We have many more benches than them, but our positions aren't that good for a mapping service where you'd want a line of benches shown as a neat line near a path and on the correct side. I think someone suggested importing from OSM to OB in the past, but we really need photos they don't provide. If OSM were comprehensive with their bench coverage, we could use them to target mapping expeditions, but I don't think they have enough, and it would be frustrating to visit a set with no plaques. When I set out doing a benching trip, I glance at OSM to look for likely locations, but their bench symbols rarely give rich pickings.

Also councils move benches around all the time, not a problem for us focussed on inscriptions, but a mapping annoyance.

**Comment by backache** [comment]
Interesting
 If the council are moving the bench around you're right, that is an issue for them, but funnily enough, we can help in that the inscription is in effect a "fingerprint" for the bench which OB then has as a unique id. This means if OB senses a bench has moved because it's been logged in a different place or moved in the db ,  OSM could be updated

**Comment by hb0nd** [comment]
A [maproulette.org](https://www.maproulette.org ) challenge would be best to conflate the data over to OSM, as it relies on manual review. Perhaps matching benches within a metre, then reviewing the images to see if it's the same bench?

**Comment by backache** [comment]
Map roulette sounds like a good idea, make it fun and it's no longer a chore
